ledger-api/grpc-definitions/BUILD.bazel

```diff
@@ -70,69 +73,115 @@ proto_gen(
     ],
 )
 
-# TODO(nic): When compile-proto-file handles map<key,value>, we will can run on:  **/*.proto
-# + glob(["**/*.proto",])
+google_protobuf_src = "external/com_google_protobuf/src"
+
+genrule(
+    name = "google-protobuf-haskellpb-sources",
+    srcs = ["@com_google_protobuf//:well_known_protos"],
+    outs = ["Google/Protobuf/" + b for b in [
+        "Any.hs",
+        "Duration.hs",
+        "Empty.hs",
+        "Timestamp.hs",
+        "Wrappers.hs",
+    ]],
+    cmd = """
+        for src in \
+            external/com_google_protobuf/src/google/protobuf/any.proto \
+            external/com_google_protobuf/src/google/protobuf/duration.proto \
+            external/com_google_protobuf/src/google/protobuf/empty.proto \
+            external/com_google_protobuf/src/google/protobuf/timestamp.proto \
+            external/com_google_protobuf/src/google/protobuf/wrappers.proto \
+        ; do
+            $(location @haskell_proto3__suite//:compile-proto-file) \
+                --includeDir """ + google_protobuf_src + """ \
+                --proto google/protobuf/$$(basename $$src) \
+                --out $(@D)
+        done
+    """,
+    tools = [
+        "@haskell_proto3__suite//:compile-proto-file",
+    ],
+)
+
+google_rpc_src = "external/com_github_googleapis_googleapis"
+
+genrule(
+    name = "google-rpc-haskellpb-sources",
+    srcs = [
+        "@com_github_googleapis_googleapis//google/rpc:status.proto",
+        "@com_google_protobuf//:well_known_protos",
+    ],
+    outs = ["Google/Rpc/Status.hs"],
+    cmd = """
+        for src in $(location @com_github_googleapis_googleapis//google/rpc:status.proto); do \
+            $(location @haskell_proto3__suite//:compile-proto-file) \
+                --includeDir """ + google_protobuf_src + """ \
+                --includeDir """ + google_rpc_src + """ \
+                --proto google/rpc/$$(basename $$src) \
+                --out $$(dirname $$(dirname $(@D))) 
+                   #2x dirname because @D works differently for a single output
+        done
+    """,
+    tools = [
+        "@haskell_proto3__suite//:compile-proto-file",
+    ],
+)
+
+v1_proto_dir = "com/digitalasset/ledger/api/v1/"
+
+v1_protos_which_use_map = [
+    # These must be excluded until we switch to upstream support for: map<key,value>
```

Comment:
Neil’s PR has been merged now so I think you might be able to get rid of this.

Reply:
yes, I did this already!